com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -249,39 +247,58 @@ private void OnValidate()
                         }
                     }
-                }
-            }
+                    else
+                    {
+                        // Defautlt to the standard NetworkPrefab.Prefab's NetworkObject first
+                        var globalObjectIdHash = networkObject.GlobalObjectIdHash;
 
-            int playerPrefabCount = NetworkConfig.NetworkPrefabs.Count(x => x.IsPlayer);
+                        // Now check to see if it has an override
+                        switch (NetworkConfig.NetworkPrefabs[i].Override)
+                        {
+                            case NetworkPrefabOverride.Prefab:
+                                {
+                                    if (NetworkConfig.NetworkPrefabs[i].SourcePrefabToOverride == null && NetworkConfig.NetworkPrefabs[i].Prefab != null)
+                                    {
+                                        NetworkConfig.NetworkPrefabs[i].SourcePrefabToOverride = NetworkConfig.NetworkPrefabs[i].Prefab;
+                                    }
+                                    globalObjectIdHash = NetworkConfig.NetworkPrefabs[i].SourcePrefabToOverride.GetComponent<NetworkObject>().GlobalObjectIdHash;
+                                }
+                                break;
+                            case NetworkPrefabOverride.Hash:
+                                globalObjectIdHash = NetworkConfig.NetworkPrefabs[i].SourceHashToOverride;
+                                break;
+                        }
 
-            if (playerPrefabCount == 0 && !NetworkConfig.ConnectionApproval && NetworkConfig.CreatePlayerPrefab)
-            {
-                if (NetworkLog.CurrentLogLevel <= LogLevel.Normal)
-                {
-                    NetworkLog.LogWarning($"There is no {nameof(NetworkPrefab)} marked as a {nameof(NetworkPrefab.IsPlayer)}");
-                }
-            }
-            else if (playerPrefabCount > 1)
-            {
-                if (NetworkLog.CurrentLogLevel <= LogLevel.Normal)
-                {
-                    NetworkLog.LogWarning($"Only one {nameof(NetworkPrefab)} can be marked as a {nameof(NetworkPrefab.IsPlayer)}");
+                        // Add to the NetworkPrefabOverrideLinks or handle a new (blank) entries
+                        if (!NetworkConfig.NetworkPrefabOverrideLinks.ContainsKey(globalObjectIdHash))
+                        {
+                            NetworkConfig.NetworkPrefabOverrideLinks.Add(globalObjectIdHash, NetworkConfig.NetworkPrefabs[i]);
+                        }
+                        else
+                        {
+                            // This happens when a new duplicate entry is created.
+                            // We just turn it into a new blank entry
+                            NetworkConfig.NetworkPrefabs[i] = new NetworkPrefab();
+                        }
```

Comment:
I actually didn't quite get this block.
The first if block looks sensible but the second else block made me confused a little bit :/
Do you want to explain a little bit?

Reply:
The default behavior for adding new list entries in Unity is that the previous list entry (if any) is basically duplicated.
Without adding additional custom UI elements and/or properties, this basically assures that any duplicate entry and/or any new entry will be a new, empty, NetworkPrefab.
As the comments suggest... :)